demo/server/src/modules/datasync/MongoDataHandler.ts

```diff
@@ -0,0 +1,99 @@
+import { sync } from '@raincatcher/datasync-cloud';
+import { Db } from 'mongodb';
+
+/**
+ * Initializes global data handlers for feedhenry sync
+ */
+export class GlobalMongoDataHandler {
+  constructor(private db: Db) {
+  }
+
+  /**
+   * Init all handlers for CRUD operations
+   */
+  public initGlobalHandlers() {
+    this.handleList();
+    this.handleCreate();
+    this.handleSave();
+    this.handleCreate();
+    this.handleDelete();
+  }
+
+  public handleList() {
+    const self = this;
+    sync.globalHandleList(function(datasetId, queryParams, metadata, cb) {
+      queryParams = queryParams || {};
+      const resultPromise = this.db.collection(datasetId).find(queryParams);
+      return resultPromise.toArray().then(function(list: any[]) {
+        const syncData: any = {};
+        list.forEach(function(object) {
+          syncData[object.id] = object;
+        });
+        return cb(undefined, syncData);
+      }).catch(function(err: any) {
+        return cb(err);
+      });
+    });
+  }
+
+  public handleCreate() {
+    const self = this;
+    sync.globalHandleCreate(function(datasetId, data, metadata, cb) {
+      self.db.collection(datasetId).insertOne(data).then(function() {
+        const res = {
+          uid: data.id,
+          data
+        };
+        return cb(undefined, res);
```

Comment:
WDYT about shortening this type of function to `.then(() => cb(undefined, { uid: data.id, data }))` ?